### ISSUE TITLE
Lots of inaccuracies

### ISSUE THREAD

**Issue by TheGermanEngie** [mirix/approaches-to-diarisation#9]

**Comment by mirix** [comment]
I don't know. I have tested it in nearly 60 samples so far and it produces pretty good results in all except 4.

I am currently working in handling those few exceptions. I expect to release a more robust version next week or so. 

It will probably be based upon faster_whisper and not stable_ts anymore. So it should be faster.

Have you had a look to the clusters (the HTML files)? Perhaps that will give you some clues. 

Check the UMAP and HDBSCAN parameters. Namely, for UMAP, n_neighbors and, for HDBSCAN, cluster_size and cluster_selection_epsilon.

I am sure that tweaking those parameters will give you what you want for that particular sample, the challenge is to automatically find the optimal parameters for each sample. I will need to develop some heuristic algorithms, but I do not have the time right now.

In my case, it uses one GPU even if two are available. But keep in mind that, if you have a CUDA accelerator, the most time-consuming tasks are going to be clustering-related and neither UMAP or HDBSCAN can benefit from GPU acceleration at the moment.

**Comment by mirix** [comment]
New pipeline:

https://github.com/mirix/approaches-to-diarisation/issues/10#issue-1873161024

**Comment by TheGermanEngie** [comment]
Nice, will get back to soon.

**Comment by mirix** [comment]
I fixed a bug and uploaded the new script right now.

The main issue I have with WhisperX is that it is unable to align non-dictionary tokens, which, essentially, makes it useless for us.

**Comment by dgoryeo** [comment]
Hi @mirix , have you had a chance to test this with Japanese and Chinese? 
Thanks

**Comment by mirix** [comment]
@dgoryeo No I haven't. At this stage we are mostly interested in English, French and German. If my memory serves me well, Whisper does support Japanese and Chinese. The issue is that Stable Whisper and the subsequent post-processing rely strongly on punctuation rules. Therefore, at least those modules will need to be modified somehow.

**Comment by dgoryeo** [comment]
You're absolutely right @mirix . My experience with stable-ts, WhisperX, and WhisperHallu has been that when it comes to Japanese and Chinese the post processing falls short.

**Comment by mirix** [comment]
@dgoryeo Perhaps you can give a try to Alibaba's FunASR.

**Comment by mirix** [comment]
There is also Paddle: https://github.com/PaddlePaddle/PaddleSpeech

I personally haven't tried any of the two, but they seem geared primarily towards Chinese.

**Comment by dgoryeo** [comment]
PaddleSpeech looks easier --it has CLI. I'l start with that. Thanks @mirix !
